File: java/rlib/TestingUtils.java
package rlib;

public class TestingUtils {

  public static class Expectation {

    private Object result;

    public Expectation(Object result) {
      this.result = result;
    }

    public void toBe(Object reference) {
      if (reference == null) {
        if (this.result == null) return;
        else {
          System.out.printf("Expected %s to be \"%s\".\n", this.result.toString(), "null");
        }
      } else {
        if (reference.equals(this.result)) return;
        else {
          System.out.printf("Expected %s to be \"%s\".\n", "null", reference.toString());
        }
      }
    }

    public void toNotBe(Object reference) {
      if (reference == null) {
        if (this.result != null) return;
        else {
          System.out.printf("Expected %s to not be \"%s\".\n", this.result.toString(), "null");
        }
      } else {
        if (!reference.equals(this.result)) return;
        else {
          System.out.printf("Expected %s to not be \"%s\".\n", "null", reference.toString());
        }
      }
    }

  }

  public static Expectation expect(Object result) {
    return new Expectation(result);
  }

}

File: python/rlib/testing_utils.py
#!/usr/bin/env python3

class Expectation():

    def __init__(self, result):
        self.result = result

    def to_be(self, reference):
        if self.result != reference:
            print('Expected "%s" to be "%s".' % (self.result, reference))

    def to_not_be(self, reference):
        if self.result == reference:
            print('Expected "%s" to not be "%s".' % (self.result, reference))

    def to_pass(self, checker, *args):
        if not checker(self.result, args):
            print('Expected "%s" to pass "%s".' % (self.result, checker))

def expect(result):
    return Expectation(result)
